File: src/test/java/com/widehouse/whatnext/domain/CategoryTest.java
package com.widehouse.whatnext.domain;

import static org.assertj.core.api.BDDAssertions.then;

import org.junit.Test;

public class CategoryTest {
    @Test
    public void constructorTest() {
        Category category = new Category(1, "home", "ff0000");

        then(category)
                .hasFieldOrPropertyWithValue("id", 1)
                .hasFieldOrPropertyWithValue("name", "home")
                .hasFieldOrPropertyWithValue("color", "ff0000");
    }
}

File: src/test/kotlin/com/widehose/whatnext/domain/CategoryTest.kt
package com.widehose.whatnext.domain

import com.widehouse.whatnext.domain.Category
import org.assertj.core.api.BDDAssertions
import org.junit.Test


class CategoryTest {
    @Test
    fun constructorTest() {
        val tested = Category(1, "home", "ffffff").apply {
            BDDAssertions.then(this)
                    .hasFieldOrPropertyWithValue("id", 1)
                    .hasFieldOrPropertyWithValue("name", "home")
                    .hasFieldOrPropertyWithValue("color", "ffffff")

            id = 2
            BDDAssertions.then(id).isEqualTo(2)
            name = "work"
            BDDAssertions.then(name).isEqualTo("work")
            color = "000000"
            BDDAssertions.then(color).isEqualTo("000000")
        }
    }
}